core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -54,18 +60,38 @@ open class RestActionTestCaseNamingStrategy(
      * following solvers.
      */
     override fun resolveAmbiguities(duplicatedIndividuals: Set<EvaluatedIndividual<*>>): Map<EvaluatedIndividual<*>, String> {
-        val solvedAmbiguities = mutableMapOf<EvaluatedIndividual<*>, String>()
         val workingCopy = duplicatedIndividuals.toMutableSet()
 
-        val pathDisambiguatedIndividuals = solvePathAmbiguities(workingCopy)
-        solvedAmbiguities.putAll(pathDisambiguatedIndividuals)
-        removeSolvedDuplicates(workingCopy, pathDisambiguatedIndividuals.keys)
+        val ambiguitySolversPerIndividual = mutableMapOf<EvaluatedIndividual<*>, MutableList<(Action) -> List<String>>>()
 
-        val queryParamsDisambiguatedIndividuals = solveQueryParamsAmbiguities(workingCopy)
-        solvedAmbiguities.putAll(queryParamsDisambiguatedIndividuals)
-        removeSolvedDuplicates(workingCopy, queryParamsDisambiguatedIndividuals.keys)
+        val pathDisambiguatedIndividuals = getPathDisambiguationIndividuals(workingCopy)
+        pathDisambiguatedIndividuals.forEach {
+            ambiguitySolversPerIndividual[it] = mutableListOf(::pathAmbiguitySolver)
+        }
+
+        val queryParamsDisambiguatedIndividuals = getQueryParamsDisambiguationIndividuals(workingCopy)
+        queryParamsDisambiguatedIndividuals.forEach {
+            if (ambiguitySolversPerIndividual.containsKey(it)) {
+                ambiguitySolversPerIndividual[it]?.add(::queryParamsAmbiguitySolver)
+            } else {
+                ambiguitySolversPerIndividual[it] = mutableListOf(::queryParamsAmbiguitySolver)
+            }
+        }
 
-        return solvedAmbiguities
+        return collectSolvedNames(ambiguitySolversPerIndividual)
+    }
+
+    /*
+     * When two or more individuals share a name, no disambiguation is performed.
+     * Otherwise, we would just be increasing test case name length without having actually disambiguated.
+     */
+    private fun collectSolvedNames(ambiguitySolversPerIndividual: MutableMap<EvaluatedIndividual<*>, MutableList<(Action) -> List<String>>>): Map<EvaluatedIndividual<*>, String> {
```

Comment:
as you are not planning to modify `ambiguitySolversPerIndividual` in this function, it should be passed as a `Map` and not a `MutableMap`